src/onelogin/api/client.py

```diff
@@ -158,11 +159,18 @@ def retrieve_apps_from_xml(self, xml_content):
     def is_expired(self):
         return self.expiration is not None and datetime.datetime.now(tz.tzutc()) > self.expiration
 
+    def last_request_forbidden(self):
```

Comment:
I wonder if a better way to do this is 
* make any request
* if the request fails due to 401 then the access token is no good so attempt to fetch a new access token
    * if the access token request succeeds then retry the original request 
    * if the access token request fails then return 401 error